### ISSUE TITLE
Lockup in maven concurrent build, with fork=false

### ISSUE THREAD

**Issue by aaime** [spotbugs/spotbugs-maven-plugin#104]
I've just added spotbugs to a QA build that already had PMD and errorprone, the build times on Travis went through the roof, from 17m to 44m (parallel build, -T1C), which is dangerously close to the Travis max build time limit.

So I've tried to disable forking spotbugs hoping to get a speedup, but the build locked up after a few modules.

Here is a jstack:
[spotbugs-jstack.txt](https://github.com/spotbugs/spotbugs-maven-plugin/files/2802140/spotbugs-jstack.txt)

**Comment by aaime** [comment]
Ah, tried a trick we used to speed up surefire tests, that is, setting the following JVM options:

```
   <jvmArgs>-XX:+TieredCompilation -XX:TieredStopAtLevel=1</jvmArgs>
```

and build times went down from 44m to 29m... quite a bit better :-D
Do you by any chance know of other args that we can use to speed up the spotbugs checks?

**Comment by hazendaz** [comment]
I don't know of any args to make this faster.  I did user the settings you noted there on our physical build which sped up travis for us by about 8 minutes overall.  However we are running quite a few integration tests.  Our build takes 28 minutes on travis running in two separate jdks.  You might want to take a look at what we are doing there to see if that helps shed any insite for you.

**Comment by hazendaz** [comment]
This is likely fixed by #535

**Comment by hazendaz** [comment]
closing as this issue is believed fixed.  If still an issue, please open a new ticket based on current data.
